### PR TITLE
feat: per-package-manager `dependencies` block with scoped trust

### PR DESCRIPTION
**Pull request by roxblnfk**
## What

The flat trust surface (\`trusted\`, \`trusted-replace\`) and the \`local\` toggle map merge into one per-package-manager block — **\`dependencies\`**:

```jsonc
{
  "dependencies": {
    "composer": { "enabled": true, "trusted": ["acme/*"], "trusted-replace": false },
    "npm":      { "trusted": ["@myorg/*"] },
    "go":       true
  }
}
```

## Why

- **Trust is ecosystem-scoped in reality.** The built-in list is already per-manager (\`resources/trusted-composer.txt\`, resolved by provider id), and the \`vendor/pkg\` pattern grammar cannot express npm names (\`@scope/pkg\`) or Go module paths (\`github.com/owner/mod\`). A per-manager \`trusted\` list is a precondition for the npm/go providers, not a style choice.
- **\`local\` stopped describing its content** the moment it grew past on/off toggles: the block configures how the project's *dependencies* become skill donors — the key now says that.

## Semantics

- Value is \`bool | object\`: \`true\` ≡ \`{"enabled": true}\`. \`enabled\` absent → per-manager default (composer on, npm/go off until their providers land). Configuring \`trusted\` does **not** implicitly enable a manager.
- Per-manager grammar: composer patterns validate as today; npm/go lists are locked structurally now, grammar-enforced when their providers ship.
- \`sources[]\` entries are unaffected — they are trusted by declaration and never consult these lists.

## Migration (same machinery as \`remote\` → \`sources\`)

- Write-mode commands restructure a legacy file in place: \`local\` map + flat trust keys fold into \`dependencies\` at the first legacy key's position, everything else byte-preserved, \`[migrate]\` notice. A fully-legacy file (\`remote\` + \`trusted\` + \`local\`) gets both migrations in one run.
- Read-only surfaces read the aliases and emit a \`[deprecated]\` notice; they never rewrite.
- Mixing \`dependencies\` with any legacy key in one block is a fatal config error — one security surface, one form.
- The interactive \`skills:init --force\` path folds legacy defaults before prompting (review caught a trust-broadening regression here — a legacy \`trusted-replace: true\` would have silently flipped off; fixed with round-trip coverage).

## Runtime

\`ProjectConfig\`'s existing fields are folded from the block, so \`SyncPlanner\`, \`DonorProviderBuilder\`, trust attribution and every downstream stage are untouched — behaviour parity between the flat form and its translation is pinned by tests. The full per-manager structure rides on \`ProjectConfig::$dependencies\`, ready for the npm/go providers.

## Tests

Unit matrix (forms, defaults, grammars, mixing, folding parity, migration edge cases, wizard round-trip), schema matrix (valid/deprecated/mixed/unknown), acceptance: in-place restructure + sync parity, read-only notice with byte-identical file, mixed-form failure, new-form trust approval, \`composer: false\` toggle, double-migration in one run.

\`composer test\`: **842 passed / 1 skipped**. \`composer psalm\`, \`composer cs:diff\` clean.

## Out of scope

npm/go provider implementations (config shape is now locked for them), per-manager \`--trust\` CLI scoping, built-in \`trusted-npm.txt\`/\`trusted-go.txt\` lists, internal renames (\`LOCAL_IDS\` etc.), alias removal (next major).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Scope addendum

The built-in \`trusted-npm.txt\` / \`trusted-go.txt\` lists were originally listed as out of scope, but were pulled INTO this PR once the lists themselves were authored: the \`TrustedVendorRegistry\` now resolves them per provider id behind a \`TrustPattern\` abstraction (\`VendorPattern\` untouched for composer; new \`NpmPattern\`/\`GoPattern\` with per-ecosystem grammars, go wildcards using arbitrary-depth prefix semantics). Runtime consultation still arrives with the npm/go providers — the lists are wired and tested, not yet queried.

### CODE REVIEW

File: src/Init/InteractiveInitWizard.php [hunk 1]
```diff
@@ -89,11 +89,20 @@ public function run(IOInterface $io, array $defaults): ?array
...
-            $result['trusted'] = $trusted;
+            $composer['trusted'] = $trusted;
         }
         if ($trustedReplace) {
-            $result['trusted-replace'] = true;
+            $composer['trusted-replace'] = true;
+        }
+        if ($composer !== []) {
+            $result['dependencies'] = ['composer' => $composer];
         }
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply src/Init/InteractiveInitWizard.php]: Fixed in adbfce7. The wizard now seeds its result from the normalised defaults and overrides only the keys it owns; `dependencies.composer.enabled` (including an explicit `false`) and sibling manager entries (`npm`, `go`) survive a `--force` re-run. Trust answers merge at the composer-object level. Covered by a rich-file round-trip test.

File: src/Init/InteractiveInitWizard.php [hunk 2]
```diff
@@ -89,11 +89,20 @@ public function run(IOInterface $io, array $defaults): ?array
...
         if ($discovery) {
             $result['discovery'] = true;
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply src/Init/InteractiveInitWizard.php:111]: Fixed in adbfce7 — same root cause as the dependencies drop: the result was built from scratch. Unprompted keys (`sources`, `path-from-root`, …) now pass through verbatim in canonical key order, and the defaults normalisation folds legacy `remote` → `sources` first (rule extracted into a shared `ProjectConfigMigrator::renameSourcesInMap()` so one implementation owns it). This also closes the long-standing pre-existing "wizard drops sources on --force" issue.

File: src/Config/NpmPattern.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                $pattern,
+            ));
+        }
+        if ($pattern === '*') {
+            throw new \InvalidArgumentException(
+                'Invalid npm pattern "*": a bare wildcard is not allowed; scope it as "@scope/*".',
+            );
+        }
+
+        return new self(raw: $pattern, scope: null, package: $pattern);
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply src/Config/NpmPattern.php]: Fixed in adbfce7: any `*` in an unscoped name is now a load-time error (`Invalid npm pattern "lodash*": an unscoped package name cannot contain "*"; scope it as "@scope/*".`).

File: src/Config/NpmPattern.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            /** @var non-empty-string $scope the `@` guarantees a non-empty segment */
+            $scope = \substr($pattern, 0, $slash);
+            /** @var non-empty-string $package */
+            $package = \substr($pattern, $slash + 1);
+
+            return new self(
+                raw: $pattern,
+                scope: $scope,
+                package: $package === '*' ? null : $package,
+            );
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply src/Config/NpmPattern.php:109]: Fixed in adbfce7: empty scope (`@/*`), `*` in the scope segment, and `*` inside a package segment are all load-time errors now; the package segment must be exactly `*` or contain no `*`.

File: src/Config/GoPattern.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        }
+
+        if ($pattern === '*') {
+            throw new \InvalidArgumentException(
+                'Invalid go pattern "*": a bare wildcard is not allowed; give it a prefix ("owner/*").',
+            );
+        }
+
+        return new self(raw: $pattern, base: $pattern, wildcard: false);
+    }
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply src/Config/GoPattern.php:94]: Fixed in adbfce7: `*` is only accepted as the trailing `/*` wildcard; `github.com/*/mod`, `github.com/owner/*/v2`, and a wildcard base containing `*` all fail with `"*" is only allowed as the trailing "/*" wildcard.`

File: resources/trusted-npm.txt [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Built-in trusted-vendors list for the npm provider.
```
[the remark this answers is not in the document]
Comment (roxblnfk) [reply resources/trusted-npm.txt:1]: Done — the PR description now carries a scope addendum explaining why the lists were pulled into scope and what exactly is wired (registry + patterns + tests; runtime consultation still ships with the providers).
